### ISSUE TITLE
CalledProcessError saving icns file to io.BytesIO()

### ISSUE THREAD

**Issue by jeffre** [python-pillow/Pillow#4732]
Currently, writing an icns file to an io.ByteIO object fails. This operation is supported by other filetypes (jpg, png, ico, etc) so I'm not clear if this report falls under feature request or bug

### What did you do?
Write icns file to io.BytesIO() object
```python
from PIL import Image
import io
import base64

im = Image.new(mode="RGBA", size=(256,256), color=(0,100,246,255))
f = io.BytesIO()
im.save(fp=f, format="icns")

b64 = base64.b64encode(f.getvalue()).decode()
print(f'Image: {b64}')
```

### What did you expect to happen?
Fill the io.BytesIO object 

### What actually happened?
macOS's iconutil failed because it was provided an empty filename for output (-o)
```bash
subprocess.CalledProcessError: Command '['iconutil', '-c', 'icns', '-o', '', '/var/folders/91/q75bkgks1196cr1rsr0399540000gn/T/tmppmsm3z2y.iconset']' returned non-zero exit status 1.
```

I believe one solution would be if `filename` variable is empty then use a dummy name and after popen returns, read the file from disk.

### What are your OS, Python and Pillow versions?

* OS: macOS Mojave 10.14.6
* Python: 3.7.7
* Pillow: 7.1.2

**Comment by nulano** [comment]
This would be fixed by merging #4526.

**Comment by radarhere** [comment]
I've created #4741 to fix this, in case it makes it in before #4526.
